Assets/BossRoom/Scripts/Client/Game/Action/FXProjectileTargetedActionFX.cs

```diff
@@ -92,7 +92,7 @@ private void PlayHitReact()
             if (m_Target && m_Target.TryGetComponent(out Client.ClientCharacter clientCharacter) && clientCharacter.ChildVizObject != null )
             {
                 var hitReact = !string.IsNullOrEmpty(Description.ReactAnim) ? Description.ReactAnim : k_DefaultHitReact;
-                clientCharacter.ChildVizObject.OurAnimator.SetTrigger(hitReact);
+                clientCharacter.ChildVizObject.TrySetTrigger(hitReact);
```

Comment:
oh, I just saw the TrySetTrigger implementation. So this isn't just client code, it's also authoritative code, correct?
So this works because we have a host that's both a client and a server?
I would have expected the set trigger stuff to happen in the "server" part of this action, no? Since we're server authoritative, I would expect the information sending to happen from server code.

Reply:
So yes that's how this is working. It was the "safer" implementation. What you're suggesting would mean moving some of the client-side animation code to the server-side Actions, which I agree is the place where animations should play, since we're establishing server authority.

I took one more peek at when animations play on clients and they're for the most part triggered when ActionFX's start, end, or hit something. So it seems like it can be converted to server-side code.

One last thing to point out is that we'll still have some actions that play anticipated animations, so we'll still have a few animations playing on client-side ActionFXs.

Reply:
for animation anticipation, discussed this with Phil, there could be a way to inherit from NetworkAnimator and have a way to add some API for anticipation.
This could be done samples side

Reply:
Following up on this specific item offline.